tools/run_tests/performance/loadtest_examples.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+# Copyright 2021 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script generates a set of load test examples from templates.
+
+LOADTEST_CONFIG=tools/run_tests/performance/loadtest_config.py
+
+if (( $# < 1 )); then
+    echo "Usage: ${0} <output directory>"
+    exit 1
+fi
+
+if [[ ! -x "${LOADTEST_CONFIG}" ]]; then
+    echo "${LOADTEST_CONFIG} not found."
+    exit 1
+fi
+
+outputbasedir="${1}"
+
+mkdir -p "${outputbasedir}/templates"
+
+example_file() {
+    local scenario="${1}"
+    local suffix="${2}"
+    if [[ "${scenario#cpp_}" != "${scenario}" ]]; then
+        echo "cxx${suffix}"
+        return
+    fi
+    if [[ "${scenario#python_asyncio_}" != "${scenario}" ]]; then
+        echo "python_asyncio${suffix}"
+        return
+    fi
+    echo "${scenario%%_*}${suffix}"
+}
+
+example_language() {
+    local filename="${1}"
+    if [[ "${filename#cxx_}" != "${filename}" ]]; then
+        echo "c++"
+        return
+    fi
+    if [[ "${filename#python_asyncio_}" != "${filename}" ]]; then
+        echo "python_asyncio"
+        return
+    fi
+    echo "${filename%%_*}"
+}
+
+scenarios=(
+    "cpp_generic_async_streaming_ping_pong_secure"
+    "csharp_protobuf_async_unary_ping_pong"
+    "go_generic_sync_streaming_ping_pong_secure"
+    "java_generic_async_streaming_ping_pong_secure"
+    "node_to_node_generic_async_streaming_ping_pong_secure"
+    "php7_protobuf_php_extension_to_cpp_protobuf_sync_unary_ping_pong"
+    "python_generic_sync_streaming_ping_pong"
+    "python_asyncio_generic_async_streaming_ping_pong"
+    "ruby_protobuf_sync_streaming_ping_pong"
+)
+
+basic_example() {
+    local -r scenario="${1}"
+    local -r outputdir="${2}"
+    local -r outputfile="$(example_file "${scenario}" _example_loadtest.yaml)"
+    local -r language="$(example_language "${outputfile}")"
+    ${LOADTEST_CONFIG} \
+        -l "${language}" \
+        -t ./tools/run_tests/performance/templates/loadtest_template_basic_all_languages.yaml \
+        -s client_pool= -s server_pool= -s big_query_table= \
```

Comment:
This looks incomplete, missing the `value` part of `-s key=value`.

Reply:
Right, what that does is set the value to an empty string. This is intended – for the "basic" examples, we are not setting pools or saving to BigQuery. The code now removes empty fields when generating the config, so the final result looks as if fields like "pool" never existed. We do add replacement keys to the "prebuilt" examples.

Reply:
Ah, ok. Is there anywhere that explains why those features should be stripped from the basic examples? I see the basic template does include placeholders for pools.

Reply:
The idea was to have the basic examples be runnable as is, while the examples for prebuilt images are templates and require some key substitution.

Added comments to the README pointing to the grpc/test-infra README, as well as comments in the script itself to clarify.